src/main/java/gov/nasa/pds/tools/validate/rule/pds4/FileReferenceValidationRule.java

```diff
@@ -350,6 +426,52 @@ private boolean validate(DocumentInfo xml) {
                       fileObject.getLineNumber(), -1));
                   passFlag = false;
                 }
+
+
+                // Check for JPEG file validity.
+                try {
+                  problems.addAll(handleJPEG(target, urlRef,
+                      fileObject, jpegName, parent, directory));
+                } catch (Exception e) {
+                  ProblemDefinition def = new ProblemDefinition(
+                      ExceptionType.ERROR, ProblemType.INTERNAL_ERROR,
+                      "Error occurred while processing JPEG file content for "
+                      + FilenameUtils.getName(urlRef.toString()) + ": "
+                      + e.getMessage());
+                  problems.add(new ValidationProblem(def, target,
+                      fileObject.getLineNumber(), -1));
+                  passFlag = false;
+                }
+
+                // Check for PNG file validity.
+                try {
+                  problems.addAll(handlePNG(target, urlRef,
+                      fileObject, pngName, parent, directory));
+                } catch (Exception e) {
+                  ProblemDefinition def = new ProblemDefinition(
+                      ExceptionType.ERROR, ProblemType.INTERNAL_ERROR,
+                      "Error occurred while processing PNG file content for "
+                      + FilenameUtils.getName(urlRef.toString()) + ": "
+                      + e.getMessage());
+                  problems.add(new ValidationProblem(def, target,
+                      fileObject.getLineNumber(), -1));
+                  passFlag = false;
+                }
+
+                passFlag = this.checkGenericDocument(target, urlRef, fileObject, htmlName, parent, directory, documentStandardId, "HTML", problems, ProblemType.NON_HTML_FILE);
```

Comment:
@qchaupds it seems like this should be a loop through a hashmap, but not worth holding up the PR right now.